### ISSUE TITLE
Feature request - Customizable Shareable URLS

### ISSUE THREAD

**Issue by SanderCN** [ShaneIsrael/fireshare#174]
Hi there!

I am currently looking into this project as I want to switch to it but one thing that is kind of holding me back is customizable shareable URLs.

What I mean by this is instead of: https://v.fireshare.net/w/99fa092699f274b6ae749bbccb339980 I can create a custom URL like: https://v.fireshare.net/w/best%20collateral

This would look way better when sharing video's across a chat for example and that's exactly why I would want that feature.

Thanks for reading :)

**Comment by ShaneIsrael** [comment]
Unfortunately the way Fireshare is designed a unique url like that wouldn't work because it would require a database entry in order to correlate to the symlink file. The URL matches the symlink file name exactly so that the internal nginx process can be used to serve the video and not require a python backend process to serve it. 

I also do not want to add in the ability to override the url name because it would break a core feature of fireshare which is that the url names are generated based off of a hash of the actual file. Meaning that if you ever deleted your fireshare database either by acciden, on purpose, or just moved fireshare to a new system it would still generate identical URL's to your videos meaning as long as you have the original video the url will always work.

If I allowed overriding that, this process would no longer work and if the fireshare database was ever moved or deleted those overridden urls would no longer work because they would require a db lookup. 

**tl;dr**
Unfortunately your request is not possible without breaking one of the main aspects of Fireshare. If that is a deal breaker for you I'm sorry. Fireshare does support open graph data. So if you share a link to your video in most modern chat programs it should display your videos title/description.
